unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxConfiguration.cs

```diff
@@ -0,0 +1,844 @@
+            selectedMat.SetColor("_color_" + layerNum, layer.color.Evaluate(normalizedLayerTime));
+            selectedMat.SetVector("_tilingAndOffset_" + layerNum, new Vector4(layer.particleTiling.x, layer.particleTiling.y, layer.particlesOffset.x, layer.particlesOffset.y));
+            selectedMat.SetVector("_speedAndRotation_" + layerNum, GetTransitionValue(layer.particleRotation, normalizedLayerTime * 100));
+            selectedMat.SetVector("_particlesMainParameters_" + layerNum, new Vector4(layer.flipbookAnimSpeed, layer.particlesAmount, layer.particleMinSize, layer.particleMaxSize));
+            selectedMat.SetVector("_particlesSecondaryParameters_" + layerNum, new Vector4(layer.particlesHorizontalSpread, layer.particlesVerticalSpread, layer.particleMinFade, layer.particleMaxFade));
+
+
+        }
+
+        #endregion
+
+        #region Transition Values Utility Methods
+
+        float GetTransitionValue(List<TransitioningFloat> _list, float percentage, float defaultVal = 0)
+        {
+            if (_list == null || _list.Count < 1)
+            {
+                return defaultVal;
+            }
+
+            if (_list.Count == 1)
+            {
+                return _list[0].value;
+            }
+
+            TransitioningFloat min = _list[0], max = _list[0];
+
+
+            for (int i = 0; i < _list.Count; i++)
+            {
+                if (percentage <= _list[i].percentage)
+                {
+                    max = _list[i];
+
+                    if ((i - 1) > 0)
+                    {
+                        min = _list[i - 1];
+                    }
+
+                    break;
+                }
+            }
+
+            float t = Mathf.InverseLerp(min.percentage, max.percentage, percentage);
+            return Mathf.Lerp(min.value, max.value, t);
+        }
+
+        Vector2 GetTransitionValue(List<TransitioningVector2> _list, float percentage, Vector2 defaultVal = default(Vector2))
+        {
+            Vector2 offset = defaultVal;
+
+            if (_list == null || _list.Count == 0)
+            {
+                return offset;
+            }
+
+            if (_list.Count == 1)
+            {
+                offset = _list[0].value;
+                return offset;
+            }
+
+
+            TransitioningVector2 min = _list[0], max = _list[0];
+
+            for (int i = 0; i < _list.Count; i++)
+            {
+                if (percentage <= _list[i].percentage)
+                {
+                    max = _list[i];
+
+                    if ((i - 1) > 0)
+                    {
+                        min = _list[i - 1];
+                    }
+
+                    break;
+                }
+            }
+            float t = Mathf.InverseLerp(min.percentage, max.percentage, percentage);
+            offset = Vector2.Lerp(min.value, max.value, t);
+
+            return offset;
+        }
+
+        Vector3 GetTransitionValue(List<TransitioningVector3> _list, float percentage)
```

Comment:
```suggestion
        Vector3 GetTransitionValue(List<TransitioningVector3> list, float percentage)
```
We don't use underscoring for private fields or arguments, we should remove all occurrences to keep consistence across the project.